flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/FailoverStrategyLoader.java

```diff
@@ -63,7 +63,7 @@
 					return new RestartAllStrategy.Factory();
 
 				case PIPELINED_REGION_RESTART_STRATEGY_NAME:
```

Comment:
I think the user should be able to keep using the legacy `RestartPipelinedRegionStrategy` for now.

Reply:
Shall we make AdaptedRestartPipelinedRegionStrategyNG as default "region" failover strategy and add a "region_legacy" for RestartPipelinedRegionStrategy?
As the legacy RestartPipelinedRegionStrategy is not applicable for batch jobs.

Or adding  a "region_new" for AdaptedRestartPipelinedRegionStrategyNG and let users to select it when running batch jobs?

Reply:
Here is what I think:
- default should be `AdaptedRestartPipelinedRegionStrategyNG`
- `region-legacy` will use `RestartPipelinedRegionStrategy`
- `region` will use `AdaptedRestartPipelinedRegionStrategyNG`

User will get the best experience out of the box but can switch to old behavior in case of bugs.

Reply:
Ok. Let's take this way.

Reply:
If it's too much work to get all tests running with `AdaptedRestartPipelinedRegionStrategyNG` as the default, I would be ok with leaving it to "restart all". We will have to fix tests in a separate PR.

Reply:
Ok. Let's make the adapted region failover strategy default in another PR.